cluster-autoscaler/cloudprovider/gce/templates_test.go

```diff
@@ -127,15 +139,36 @@ func TestBuildNodeFromTemplateSetsResources(t *testing.T) {
 				"DNS_SERVER_IP: '10.0.0.10'\n" +
 				"AUTOSCALER_ENV_VARS: os_distribution=cos;os=linux;kube_reserved=cpu=0,memory=0,ephemeral-storage=0;BLOCK_EPH_STORAGE_BOOT_DISK=false\n" +
 				"NODE_TAINTS: 'dedicated=ml:NoSchedule,test=dev:PreferNoSchedule,a=b:c'\n",
-			physicalCpu:               8,
-			physicalMemory:            200 * units.MiB,
-			physicalEphemeralStorage:  300,
-			reservedCpu:               "0m",
-			reservedMemory:            fmt.Sprintf("%v", 0*units.MiB),
-			reservedEphemeralStorage:  "0Gi",
-			kubeReserved:              true,
-			isEphemeralStorageBlocked: false,
-			expectedErr:               false,
+			reservedCpu:              "0m",
+			reservedMemory:           fmt.Sprintf("%v", 0*units.MiB),
+			reservedEphemeralStorage: "0Gi",
+			kubeReserved:             true,
+			expectedErr:              false,
+		},
+		{
+			scenario:                      "more local SSDs requested for ephemeral storage than attached",
+			kubeEnv:                       "AUTOSCALER_ENV_VARS: os_distribution=cos;os=linux;ephemeral_storage_local_ssd_count=1\n",
+			ephemeralStorageLocalSSDCount: 1,
+			attachedLocalSSDCount:         0,
+			expectedErr:                   true,
+		},
+		{
+			scenario:                      "all attached local SSDs requested for ephemeral storage",
+			kubeEnv:                       "AUTOSCALER_ENV_VARS: os_distribution=cos;os=linux;ephemeral_storage_local_ssd_count=2\n",
+			physicalCpu:                   8,
+			physicalMemory:                200 * units.MiB,
+			ephemeralStorageLocalSSDCount: 2,
+			attachedLocalSSDCount:         2,
+			expectedErr:                   false,
+		},
+		{
+			scenario:                      "more local SSDs attached than requested for ephemeral storage",
+			kubeEnv:                       "AUTOSCALER_ENV_VARS: os_distribution=cos;os=linux;ephemeral_storage_local_ssd_count=2\n",
+			physicalCpu:                   8,
+			physicalMemory:                200 * units.MiB,
+			ephemeralStorageLocalSSDCount: 2,
+			attachedLocalSSDCount:         4,
+			expectedErr:                   false,
 		},
```

Comment:
I think we might need a test case with ephemeral storage in kube_reserved and localSSD.

Can you confirm if kube_reserved is using LocalSSDs as ephemeral storage and not boot disk in the current case

Reply:
I confirm. Adding a test...